cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1611,7 +1634,9 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> json_column_to
                                                   mr);
 
       // Reset nullable if we do not have nulls
-      if (col->null_count() == 0) { col->set_null_mask(rmm::device_buffer{0, stream, mr}, 0); }
+      if (target_type.id() == type_id::STRING and col->null_count() == 0) {
```

Comment:
This is to match the existing JSON reader behaviour:
- remove null mask for string columns, if all-valid
- otherwise keep null mask